lib/tasks/grade.rake

```diff
@@ -92,6 +97,28 @@ namespace :grade do
 
 end
 
+def sync_specs_with_source
+  reponame = `basename -s .git \`git config --get remote.origin.url\``.chomp
+  full_reponame = "appdev-projects/#{reponame}"
+
+  repo_contents = Octokit.contents(full_reponame)
+  remote_spec_folder = repo_contents.find { |git_object| git_object[:name] == 'spec' }
+  remote_sha = remote_spec_folder[:sha]
+  # Discard unstaged changes in spec folder
+  `git checkout spec -q`
```

Comment:
This is nice!

I just had another idea, in order to deal with if there are any new, unstaged files in `spec/`:

- `git commit -m "Committing staged changes"`
- `git add spec/`
- `git stash`
- `git reset HEAD^ --soft`
- At this point I _think_ the `spec/` folder should have no untracked files. Now we can continue with what you have...
    - `git checkout spec -q`
    - `git clean spec -f -q`
    - etc
    
Just a thought, maybe we don't need to worry about untracked files. But, someone nefarious (I don't really believe any of our students would) could add a spec file with one trivially passing spec that's worth 1,000,000 points.

Reply:
Interesting 🤔

> git stash push -- spec/

What does `--` mean in this command? [I just recently learned about `git stash push`](https://jelani.dev/2022/10/13/git-stash-custom-path/), but I haven't seen it used with `--` before.

> Just a thought, maybe we don't need to worry about untracked files. But, someone nefarious (I don't really believe any of our students would) could add a spec file with one trivially passing spec that's worth 1,000,000 points.

I think the [`git clean -f`](https://github.com/firstdraft/grade_runner/pull/57#discussion_r994984241) will remove any untracked files. I did test a situation of creating a new spec file worth 1000 points and `git clean -f` removed if when it was untracked. If the file is committed, it's removed when the upstream copy is checked out.

https://github.com/firstdraft/grade_runner/blob/cc4b68a72c5793d273248d7edfd0a80414d7dd9a/lib/tasks/grade.rake#L109